photutils/detection/findstars.py

```diff
@@ -376,6 +377,135 @@ def find_stars(self, data):
+        The data value or pixel-wise data values to be used for the
+        detection threshold.  A 2D ``threshold`` must have the same
+        shape as ``data``.  See `detect_threshold` for one way to create
+        a ``threshold`` image.
+
+    box_size : scalar or tuple, optional
+        The size of the local region to search for peaks at every point
+        in ``data``.  If ``box_size`` is a scalar, then the region shape
+        will be ``(box_size, box_size)``.  Either ``box_size`` or
+        ``footprint`` must be defined.  If they are both defined, then
+        ``footprint`` overrides ``box_size``.
+
+    footprint : `~numpy.ndarray` of bools, optional
+        A boolean array where `True` values describe the local footprint
+        region within which to search for peaks at every point in
+        ``data``.  ``box_size=(n, m)`` is equivalent to
+        ``footprint=np.ones((n, m))``.  Either ``box_size`` or
+        ``footprint`` must be defined.  If they are both defined, then
+        ``footprint`` overrides ``box_size``.
+
+    mask : array_like, bool, optional
+        A boolean mask with the same shape as ``data``, where a `True`
+        value indicates the corresponding element of ``data`` is masked.
+
+    border_width : bool, optional
+        The width in pixels to exclude around the border of the
+        ``data``.
+
+    npeaks : int, optional
+        The maximum number of peaks to return.  When the number of
+        detected peaks exceeds ``npeaks``, the peaks with the highest
+        peak intensities will be returned.
+
+    subpixel : bool, optional
+        If `True`, then a cutout of the specified ``box_size`` or
+        ``footprint`` will be taken centered on each peak and fit with a
+        2D Gaussian (plus a constant).  In this case, the fitted local
+        centroid and peak value (the Gaussian amplitude plus the
+        background constant) will also be returned in the output table.
+
+    error : array_like, optional
+        The 2D array of the 1-sigma errors of the input ``data``.
+        ``error`` is used only to weight the 2D Gaussian fit performed
+        when ``subpixel=True``.
+
+    wcs : `~astropy.wcs.WCS`
+        The WCS transformation to use to convert from pixel coordinates
+        to ICRS world coordinates.  If `None`, then the world
+        coordinates will not be returned in the output
+        `~astropy.table.Table`.
+    """
+
+    def __init__(self, threshold, box_size=3, footprint=None, mask=None,
+                 border_width=None, npeaks=np.inf, subpixel=True, error=None,
+                 wcs=None):
+        self.threshold = threshold
+        self.box_size = box_size
+        self.footprint = footprint
+        self.mask = mask
+        self.border_width = border_width
+        self.npeaks = npeaks
+        self.subpixel = subpixel
+        self.error = error
+        self.wcs = wcs
+
+    def __call__(self, data):
+        return self.find_stars(data)
```

Comment:
This made me realize that you can do the same thing in the `StarFinderBase` class as you've done in the grouping classes: move this ``__call__`` method to the *base* class, and then you don't have to have it replicated in *all* of the subclasses.  

That's not strictly only about `PeakFinder`, but I think it's OK to just do it as a commit in this PR.